src/data-workspace/use-close-right-hand-panel-on-selection-change.js

```diff
@@ -0,0 +1,19 @@
+import { useEffect } from 'react'
+import { useSearchParams } from 'react-router-dom'
+import { useRightHandPanelContext } from '../right-hand-panel/index.js'
+
+export default function useCloseRightHandPanelOnSelectionChange() {
+    const rightHandPanel = useRightHandPanelContext()
+    const [searchParams] = useSearchParams()
+
+    useEffect(
+        () => {
+            rightHandPanel.id && rightHandPanel.hide()
+        },
+        // contextSelection is part of the dependency array as we need to hide
+        // the sidebar when the selection changes. There's no need to use the
+        // context selection in the useEffect though
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        [searchParams]
+    )
+}
```

Comment:
Probably you would want to add the rightHandPanel to the dependency array here, or work around this in another way. In the current code, you could end up with a stale reference to `rightHandPanel`.